File: truevfs-ext/truevfs-ext-insight/src/main/scala/net/java/truevfs/ext/insight/I5tIoMediator.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight

/**
 * @author Christian Schlichtherle
 */
private class I5tIoMediator(subject: String) extends I5tMediator(subject) {

  override def newStats(offset: Int): I5tStatistics = new I5tIoStatistics(this, offset)
}

File: truevfs-ext/truevfs-ext-insight/src/main/java/net/java/truevfs/ext/insight/I5tIoMediator.java
/*
 * Copyright (C) 2005-2020 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.insight;

import javax.annotation.concurrent.ThreadSafe;

/**
 * A mediator for the instrumentation of the TrueVFS Kernel with JMX.
 *
 * @author Christian Schlichtherle
 */
@ThreadSafe
final class I5tIoMediator extends I5tMediator {

    I5tIoMediator(String subject) {
        super(subject);
    }

    @Override
    I5tStatsController newController(int offset) {
        return new I5tIoStatsController(this, offset);
    }
}
